### DISCUSSION TITLE
Why animation gets stuck while using mj_jac?

### DISCUSSION THREAD

**Discussion by Rukendu** [google-deepmind/mujoco#377] [Asking for Help]
I was using API with C to control a biped. It is a double pendulum with a sphere attached at the end and it has a torso. I wanted to find the jacobian of foot(sphere) wrt hip. But everytime the animation gets stuck while i hit the code. What could be the issue?

The control code is something like :
double jacp[6]={0};
double point[3]={d->sensordata[0]-d->sensordata[3],d->sensordata[1]-d->sensordata[4],d->sensordata[2]-d->sensordata[5]};
int body = 3;
mj_jac(m,d,jacp,NULL,point,body);

**Comment by yuvaltassa** [comment]
We can't help you without being able to reproduce the issue. Here's the [implementation](https://github.com/deepmind/mujoco/blob/main/src/engine/engine_support.c#L90), since you know C perhaps just [build from source](https://mujoco.readthedocs.io/en/latest/programming.html#building-mujoco-from-source) and track down the issue yourself?

**Reply by Rukendu** [reply]
I think I have messed up with the dimension of jacp[]. Anyway, thanks for sharing the links.
